src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -1218,6 +1322,34 @@ private bool TryBindProperty(string identifier, out IEdmProperty projectedProper
             return projectedProperty != null;
         }
 
+        private bool TryBindEscapeFunction()
+        {
+            ODataPathSegment previous = this.parsedSegments[this.parsedSegments.Count - 1];
+            IEdmType bindingType = null;
+            if (previous != null)
+            {
+                bindingType = (previous is EachSegment) ? previous.TargetEdmType : previous.EdmType;

```

Comment:
>(previous is EachSegment) [](start = 30, length = 25)

Do we have a test to validate this? In general, I don't think we should support an escape function (or any other function) following $each. The only thing that should validly follow $each is a bound action.

Reply:
I have removed the check of $each at this point for escape functions if it should not be supported. 

---
In reply to: [366143523](https://github.com/OData/odata.net/pull/1621#discussion_r366143523) [](ancestors = 366143523)